src/Jobs/Recruit.Vacancies.Jobs/QueueTriggers/GenerateAllEmployerDashboardsQueueTrigger.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Services.Projections;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.StorageQueue;
+using Esfa.Recruit.Vacancies.Jobs.Configuration;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Esfa.Recruit.Vacancies.Jobs.QueueTriggers
+{
+    public class GenerateAllEmployerDashboardsQueueTrigger
+    {
+        private readonly ILogger<GenerateAllEmployerDashboardsQueueTrigger> _logger;
+        private readonly RecruitWebJobsSystemConfiguration _jobsConfig;
+        private readonly IEmployerDashboardProjectionService _projectionService;
+        private string JobName => GetType().Name;
+
+        public GenerateAllEmployerDashboardsQueueTrigger(ILogger<GenerateAllEmployerDashboardsQueueTrigger> logger, RecruitWebJobsSystemConfiguration jobsConfig, IEmployerDashboardProjectionService projectionService)
+        {
+            _logger = logger;
+            _jobsConfig = jobsConfig;
+            _projectionService = projectionService;
+        }
+
+        public async Task GenerateAllEmployerDashboardsAsync([QueueTrigger(QueueNames.GenerateAllEmployerDashboardQueueName, Connection = "QueueStorage")] string message, TextWriter log)
+        {
+            const string individualJobName = "MultiEmployerDashboardGeneratorJob";
+            if (_jobsConfig.DisabledJobs.Contains(individualJobName))
```

Comment:
As this is generating multiple documents of the same type maybe we can forego this check and only keep it on more specific one document type job e.g. the NonLevyAccountsBlocker job

Reply:
Agree this job will probably be never disabled but I feel its easier to just add the ability to disable all queue based jobs then remember which ones we can/can't disable.

Reply:
👍